pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -5230,12 +5233,203 @@ spec:
+`)
+
+	expectedPrBothStatus := parse.MustParsePipelineRun(t, `
+metadata:
+  name: test-pipeline-run-results
+  namespace: foo
+  labels:
+    tekton.dev/pipeline: test-pipeline
+  annotations: {}
+spec:
+  pipelineRef:
+    name: test-pipeline
+status:
+  runs: {}
+  pipelineSpec: 
+    results:
+    - description: pipeline result
+      name: result
+      value: $(tasks.a-task.results.a-Result)
+    - description: custom task pipeline result
+      name: custom-result
+      value: $(tasks.b-task.results.b-Result)
+    tasks:
+    - name: b-task
+      taskRef:
+        name: b-task
+        kind: Task
+    - name: a-task
+      taskRef:
+        name: a-task
+        kind: Task
+  conditions:
+  - status: "True"
+    type: Succeeded
+    reason: Succeeded
+    message: "Tasks Completed: 2 (Failed: 0, Cancelled 0), Skipped: 0"
+  pipelineResults:
+  - name: result
+    value: aResultValue
+  - name: custom-result
+    value: bResultValue
+  childReferences:
+  - apiVersion: tekton.dev/v1beta1
+    kind: TaskRun
+    name: test-pipeline-run-results-task-run-b
+    pipelineTaskName: b-task
+  - apiVersion: tekton.dev/v1beta1
+    kind: TaskRun
+    name: test-pipeline-run-results-task-run-a
+    pipelineTaskName: a-task
+  taskRuns:
+    test-pipeline-run-results-task-run-a:
+      pipelineTaskName: a-task
+      status:
+        conditions:
+        - status: "True"
+          type: Succeeded
+        taskResults:
+        - name: a-Result
+          value: aResultValue
+    test-pipeline-run-results-task-run-b:
+      pipelineTaskName: b-task
+      status:
+        conditions:
+        - status: "True"
+          type: Succeeded
+        taskResults:
+        - name: b-Result
+          value: bResultValue
+`)
 
-	reconciledRun, _ := prt.reconcileRun("foo", "test-pipeline-run-different-service-accs", []string{}, false)
+	expectedPrMinimalStatus := parse.MustParsePipelineRun(t, `
+metadata:
+  name: test-pipeline-run-results
+  namespace: foo
+  labels:
+    tekton.dev/pipeline: test-pipeline
+  annotations: {}
+spec:
+  pipelineRef:
+    name: test-pipeline
+status:
+  runs: {}
+  pipelineSpec: 
+    results:
+    - description: pipeline result
+      name: result
+      value: $(tasks.a-task.results.a-Result)
+    - description: custom task pipeline result
+      name: custom-result
+      value: $(tasks.b-task.results.b-Result)
+    tasks:
+    - name: b-task
+      taskRef:
+        name: b-task
+        kind: Task
+    - name: a-task
+      taskRef:
+        name: a-task
+        kind: Task
+  conditions:
+  - status: "True"
+    type: Succeeded
+    reason: Succeeded
+    message: "Tasks Completed: 2 (Failed: 0, Cancelled 0), Skipped: 0"
+  pipelineResults:
+  - name: result
+    value: aResultValue
+  - name: custom-result
+    value: bResultValue
+  taskRuns: {}
+  childReferences:
+  - apiVersion: tekton.dev/v1beta1
+    kind: TaskRun
+    name: test-pipeline-run-results-task-run-b
+    pipelineTaskName: b-task
```

Comment:
thought `b-task` should be a `Run` -- the `kind` should be `Example`

Reply:
it should be, however in `yaml.go` if you try to create a function called `MustParseExample` and set the kind to `Example`, it complains since `Example` is not a valid type within `v1beta1`

Reply:
i think regardless of what function you use, the main problems are:
1. `apiVersion: example.dev/v0` is not a valid package when used in `yaml.go`. I have tried it before, forgot what file it routes to, however it did not allow the `example.dev` package to be run.
2. As a result of above, trying to write `kind: Example` does not work either.

Reply:
So `example.dev/v0` isn't on the `Run`, it's only on the `taskRef`. The actual `Run` is a standard `tekton.dev/v1alpha1` `Run`. See https://github.com/tektoncd/pipeline/blob/78f706bcdf100bc394de402315e910d1cdbeb91d/pkg/reconciler/pipelinerun/pipelinerun_test.go#L599 (and the `"simple custom task with minimal embedded status"` test case specifically, since you're using minimal status here) for an example of how to use `Run` in this way.

Reply:
thank you for the clarification and feedback @abayer ! I believe my latest commit should resolve this conversation.